torch/onnx/_internal/fx/passes/functionalization.py

```diff
@@ -102,15 +103,26 @@ def _run(self, *args) -> torch.fx.GraphModule:
         module = _utils.wrap_graph_module_for_node_meta_preservation(self.module)
 
         functionalized_callable = self._functionalize(module)
-        fx_mode = "symbolic" if self.enable_dynamic_axes else "fake"
-
-        graph_module = proxy_tensor.make_fx(
-            functionalized_callable,
-            decomposition_table={},
-            tracing_mode=fx_mode,
-            _allow_non_fake_inputs=True,
-            _allow_fake_constant=self.allow_fake_constant,
-        )(*args)
+
+        fake_mode: Any = self.fake_mode
+        maybe_fake_args = self._maybe_fakefy_args(fake_mode, *args)
+        if fake_mode is not None:
+            # Using existing fake mode as context, signal `make_fx` that it does not need
+            # to create a new fake mode by passing tracing_mode as "real".
+            tracing_mode = "real"
+        else:
+            # Existing fake mode not found, signal `make_fx` to create one.
+            fake_mode = contextlib.nullcontext()
+            tracing_mode = "symbolic" if self.enable_dynamic_axes else "fake"
```

Comment:
It's possible to make this to _pass.py? So we don't write it in passes?

Reply:
true there is a bit of repetition. I'm slightly leaning towards leaving as is for now. In case there might be more changes to it.